index.js

```diff
@@ -32,8 +30,22 @@ function dispatcher (handlers) {
   start.model = setModel
   start.state = getState
   start.start = start
+  start.use = useHooks
   return start
 
+  // push an object of hooks onto an array
+  // obj -> null
+  function useHooks (hooks) {
+    assert.equal(typeof hooks, 'object', 'barracks.use: hooks should be an object')
```

Comment:
`typeof [] === 'object'`

If we're looking for an object literal the only way to be positive it's that is to do the unwieldy:

`Object.prototype.toString.call(hooks) === '[object Object]'`

Hurry for a prototypical inheritance system?